sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -551,6 +551,34 @@ abstract class JdbcDialect extends Serializable with Logging {
     if (offset > 0 ) s"OFFSET $offset" else ""
   }
 
+  /**
+   * returns the SQL text for the SELECT statement.
+   *
+   * @param options - JDBC options that contains url, table and other information.
+   * @param columnList - The names of the columns or aggregate columns to SELECT.
+   * @param tableSampleClause - The table sample clause for the SELECT statement.
+   * @param whereClause - The WHERE clause for the SELECT statement.
+   * @param groupByClause - The group by clause for the SELECT statement.
+   * @param orderByClause - The order by clause for the SELECT statement.
+   * @param limitClause - The LIMIT clause for the SELECT statement.
+   * @param offsetClause - The OFFSET clause for the SELECT statement.
```

Comment:
The problem with this approach is we need to change it every time we add a new feature (think about the table sample pushdown).

Reply:
`JdbcSQLBuilder` has been created.